jennchenn/uwexplore
# Setup backend unit testing with pytest

## Implementation description

- Setup testing framework
- Add sample test for `course_service`
- Update github actions to run test on PR/merges to main if backend changes detected
- Upgrade GH actions to `v3` and add frontend basic APIClient code to trigger build



## Steps to test

1. Check if GH actions passes
2. (optional) Run test locally:
```bash
docker exec -it backend /bin/bash -c "python -m pytest"
```



## What should reviewers focus on?

- does it work?
- excuse the very messy commit history
